File: tracking-program/8087/api/models/Schedule.js
/**
 * SwitchCommand.js
 *
 * @description :: A scheduled switch command.
 */

module.exports = {

  attributes: {

    // The project that this switch command belongs to.
    project: {
      model: 'Project',
      required: true
    },

    startDate: {
      type: 'string',
      required: true
    },

    endDate: {
      type: 'string',
      required: true
    },

    // IDs of the switches that have accepted the command.
    switches: {
      type: 'json',
      defaultsTo: []
    },

    // Whether the command was cancelled before being executed.
    isDeleted: {
      type: 'boolean',
      defaultsTo: false
    },

    // Whether the command was completed
    isCompleted: {
      type: 'boolean',
      defaultsTo: false 
    },

    // IDs of the switches that have cancelled the command.
    scheduleDetail: {
      type: 'json',
      defaultsTo: []
    },

    deviceType: {
      type: 'number',
    },

    daysOfWeek: {
      type: 'json',
      defaultsTo: []
    },

    totalHoursOff: {
      type: 'number',
      allowNull: true,
    },


  },

};

File: tracking-program/8087/flask_app/app/models/schedule.py
"""Schedule model - api/models/Schedule.js"""
from app.extensions import db
from app.models.base import BaseModel


class Schedule(BaseModel):
    __tablename__ = "schedule"

    project = db.Column(db.Integer, db.ForeignKey("project.id"), nullable=False)
    startDate = db.Column(db.String(255), nullable=False)
    endDate = db.Column(db.String(255), nullable=False)
    switches = db.Column(db.JSON, default=list)
    isDeleted = db.Column(db.Boolean, default=False)
    isCompleted = db.Column(db.Boolean, default=False)
    scheduleDetail = db.Column(db.JSON, default=list)
    deviceType = db.Column(db.Integer)
    daysOfWeek = db.Column(db.JSON, default=list)
    totalHoursOff = db.Column(db.Float, nullable=True)
